### PR TITLE
[runtime] Resolve symlinks using API provided by NSString instead of Mono.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Mono's way doesn't work with CoreCLR anyway.

### CODE REVIEW

File: runtime/runtime.m [hunk 1]
```diff
@@ -1460,10 +1459,7 @@ -(void) xamarinSetFlags: (enum XamarinGCHandleFlags) flags;
 	bundle_path = [main_bundle bundlePath];
 #endif
 
-	result = mono_path_resolve_symlinks ([bundle_path UTF8String]);
-
-	x_bundle_path = strdup (result);
-	mono_free (result);
+	x_bundle_path = strdup ([[bundle_path stringByStandardizingPath] UTF8String]);
```
Comment (mandel-macaque) [inline runtime/runtime.m:1462]: That call does a lot of path checks! Had not idea.

**Review by mandel-macaque — APPROVED** [review]

File: runtime/runtime.m (hunk 1 above)
Comment (spouliot) [reply runtime/runtime.m:1462]: docs mention
> Note that the path returned by this method may still have symbolic link components in it.

which seems to be a problem only when relative paths are used. Not sure if the mono version handled that (or not) but the call to `bundlePath` (earlier) returns a full path (so it's fine).

**Review by spouliot — APPROVED** [review]
less code == 🎉

**Review by dalexsoto — APPROVED** [review]

File: runtime/runtime.m (hunk 1 above)
Comment (Therzok) [reply runtime/runtime.m:1462]: This looks neat and something we may want to use in VSMac for some path operations.

**Pull request merged**